bitcoin/examples/ecdsa-psbt-simple.rs

```diff
@@ -0,0 +1,246 @@
+            witness: Witness::default(),
+        })
+        .collect();
+
+    // The spend output is locked to a key controlled by the receiver.
+    let spend = TxOut { value: SPEND_AMOUNT, script_pubkey: address.script_pubkey() };
+
+    // The change output is locked to a key controlled by us.
+    let change = TxOut {
+        value: CHANGE_AMOUNT,
+        script_pubkey: ScriptBuf::new_p2wpkh(&pk_change.wpubkey_hash()), // Change comes back to us.
+    };
+
+    // The transaction we want to sign and broadcast.
+    let unsigned_tx = Transaction {
+        version: transaction::Version::TWO,  // Post BIP 68.
+        lock_time: absolute::LockTime::ZERO, // Ignore the locktime.
+        input: inputs,                       // Input is 0-indexed.
+        output: vec![spend, change],         // Outputs, order does not matter.
+    };
+
+    // Now we'll start the PSBT workflow.
+    // Step 1: Creator role; that creates,
+    // and add inputs and outputs to the PSBT.
+    let mut psbt = Psbt::from_unsigned_tx(unsigned_tx.clone()).expect("Could not create PSBT");
+
+    // Step 2:Updater role; that adds additional
+    // information to the PSBT.
+    let ty = PsbtSighashType::from_str("SIGHASH_ALL").expect("valid sighash type");
+    let derivation_paths = [
+        DerivationPath::from_str("m/84'/0'/0'/0/0").expect("valid derivation path"),
+        DerivationPath::from_str("m/84'/0'/0'/1/0").expect("valid derivation path"),
+    ];
+    let mut bip32_derivations = Vec::new();
+    for (idx, pk) in pk_inputs.iter().enumerate() {
+        let mut map = BTreeMap::new();
+        let fingerprint = Fingerprint::from_str(MASTER_FINGERPRINT).expect("valid fingerprint");
+        map.insert(pk.0, (fingerprint, derivation_paths[idx].clone()));
+        bip32_derivations.push(map);
+    }
+    psbt.inputs = vec![
+        Input {
+            witness_utxo: Some(utxos[0].clone()),
+            redeem_script: Some(ScriptBuf::new_p2wpkh(&wpkhs[0])),
+            bip32_derivation: bip32_derivations[0].clone(),
+            sighash_type: Some(ty),
+            ..Default::default()
+        },
+        Input {
+            witness_utxo: Some(utxos[1].clone()),
+            redeem_script: Some(ScriptBuf::new_p2wpkh(&wpkhs[1])),
+            bip32_derivation: bip32_derivations[1].clone(),
+            sighash_type: Some(ty),
+            ..Default::default()
+        },
+    ];
+
+    // Step 3: Signer role; that signs the PSBT.
+    psbt.sign(&master_xpriv, &secp).expect("valid signature");
+
+    // Step 4: Finalizer role; that finalizes the PSBT.
+    println!("PSBT Inputs: {:#?}", psbt.inputs);
+    let final_script_witness: Vec<Witness> = psbt
+        .inputs
+        .iter()
+        .enumerate()
+        .map(|(idx, input)| {
+            let sigs: Vec<_> = input.partial_sigs.values().collect();
+            let mut script_witness: Witness = Witness::new();
+            script_witness.push(&sigs[0].to_vec());
```

Comment:
In 6564ca14397678a8c0a334bef5a6049013a634c6:

`to_vec` (allocates) should be `serialize` (does not).

Also we should really rename `serialize` to `as_bytes` or something to make it clear that it's a non-allocating method.